packages/fuels-signers/src/provider.rs

```diff
@@ -141,6 +142,24 @@ impl Provider {
         }
     }
 
+    pub async fn mint_transaction(
```

Comment:
`mint_transaction` sounds like a verb. i.e., we're minting a transaction. Which is wrong. 

I'd suggest renaming it to `mint` only. Plus, add some doc-level comments here.